### ISSUE TITLE
Release 0.5.0

### ISSUE THREAD

**Issue by k0nserv** [webrtc-rs/webrtc#235]
We haven't had a release of the top level crate(`webrtc`) in some time and there are a significant number of useful changes that we should release, therefore it's time to cut 0.5.0.

The primary thing that needs doing for this to happen is creating the release notes. Especially I'd like the release notes to make it clear what breaking changes have happened since 0.4.0.

## Todo

+ [x] Create release notes
+ [x] Release all sub-crates @k0nserv 
+ [x] Bump versions of sub-crates to latest
+ [x] Merge [RTP stats PR](https://github.com/webrtc-rs/webrtc/pull/225) @k0nserv (_optional_)
+ [x] Properly fix the issue identified in https://github.com/webrtc-rs/webrtc/pull/229 @stuqdog (_optional_)

The two optional changes can be part of a later release

**Comment by stuqdog** [comment]
I wrote a mostly*-fix for the id updating issue a couple weeks ago, but foolishly forgot to add a reviewer! 🤦

It's a one-liner [here](https://github.com/melekes/webrtc/pull/1). It was originally a PR on @melekes's revert PR, but I can easily move it over to the main branch.

Tl;dr of wall of text below: the solution I wrote is very simple, can be merged immediately, and handles all cases correctly except one (which is strictly better than we had before). A perfect solution is more invasive and adds a lot of mutability to the code base that could be problematic. If the quick and easy solution is good I'll create the PR right now, if not I'll start writing the more elaborate one but it will take longer to complete. @k0nserv please let me know what you think!

*I say "mostly" because there's still one case it doesn't handle well. To quote from [here](https://github.com/webrtc-rs/webrtc/pull/229#issuecomment-1208309753): This isn't _perfect_ (e.g., we're still sad if a user wants to set id: 0 and negotiated: false), but I believe it addresses the problem cases we've been encountering. And, replacing the existing convention of 0 == null while maintaining consistency with the spec is definitely more invasive (e.g., changing ID to Option<AtomicU16> means add mutability to lots of call sites).

To elaborate slightly: as it originally stood, we had cases wrong where someone wanted to negotiate an ID of zero. With my first change, we had cases wrong where someone wanted a non-negotiated ID of not-zero. Both of these cases are handled correctly under my new change. The only case that I can think of that isn't handled correctly is one where a user wants a non-negotiated ID of 0. My concern about the more precise change (i.e., changing the ID type to `Option<AtomicU16>` where `None` means non-assigned ID), which would handle all cases, is that this adds mutability to a lot of calls. This would be a more disruptive change and I could easily imagine run into all sorts of issues if data channels are being passed between threads. Other simpler changes (e.g., setting a magic value of `MaxInt` or `-1` to map to "non-set ID", or adding an extra bool param to `DataChannelParams` to say whether or not the ID was set) would also work, but either are hacky and similarly run into a problem of "There's a specific value we don't handle correctly", or are inconsistent with the spec. 

The "mostly correct" solution is ready to be merged right now and so could easily be part of 0.5.0. I have a preference for it because it is a clear improvement over the existing case and the workload to get it in place is very low. We can add a comment to the codebase indicating that a more robust solution is TODO. The more elaborate solution I can start working on but will take longer. Once I get confirmation on what is preferred/acceptable, I'll move forward on creating the corresponding PR.

**Comment by k0nserv** [comment]
That's @stuqdog's PR merged

**Comment by k0nserv** [comment]
Released https://github.com/webrtc-rs/webrtc/releases/tag/v0.5.0 🎉
